yarn-project/aztec/src/cli/util.ts

```diff
@@ -246,3 +249,46 @@ export async function preloadCrsDataForServerSideProving(
     await Promise.all([Crs.new(2 ** 25 + 1, undefined, log), GrumpkinCrs.new(2 ** 18 + 1, undefined, log)]);
   }
 }
+
+export async function setupUpdateMonitor(
+  autoUpdateMode: SharedNodeConfig['autoUpdate'],
+  updatesLocation: URL,
+  rollupVersion: number | 'canonical',
+  publicClient: ViemClient,
+  registryContractAddress: EthAddress,
+  updateNodeConfig?: (config: object) => Promise<void>,
+) {
+  const logger = createLogger('update-check');
+  const checker = await UpdateChecker.new({
+    rollupVersion,
+    baseURL: updatesLocation,
+    publicClient,
+    registryContractAddress,
+  });
+
+  checker.on('newRollup', ({ latestRollup, currentRollup }) => {
+    if (autoUpdateMode === 'enabled') {
+      logger.info(`New rollup detected. Restarting node`, { latestRollup, currentRollup });
+      process.exit(0);
```

Comment:
We can get the `signalHandlers` from the start cmd in here to do a graceful shutdown, and only call process.exit if we timeout (see `aztec/src/cli/util.ts#installSignalHandlers`)

Reply:
This is a good point. We want to avoid any chance of things like data corruption.